### ISSUE TITLE
Testing: docker-py tests fail to build if run after integration(-cli) tests

### ISSUE THREAD

**Issue by thaJeztah** [moby/moby#39570]
See https://github.com/moby/moby/pull/39068

If the docker-py image (which is currently built _inside_ the test container, so docker-in-docker), DNS resolution fails in the build:

```
14:40:56 W: Failed to fetch http://deb.debian.org/debian/dists/jessie/InRelease  
14:40:56 
14:40:56 W: Failed to fetch http://security.debian.org/debian-security/dists/jessie/updates/InRelease  
14:40:56 
14:40:56 W: Failed to fetch http://deb.debian.org/debian/dists/jessie/Release.gpg  Temporary failure resolving 'deb.debian.org'
14:40:56 
14:40:56 W: Failed to fetch http://security.debian.org/debian-security/dists/jessie/updates/Release.gpg  Temporary failure resolving 'security.debian.org'
14:40:56 
14:40:56 W: Some index files failed to download. They have been ignored, or old ones used instead.
```

Changing the order in which tests are run (https://github.com/moby/moby/pull/39068/commits/6aafe0fd9ed34c4d91adde62c86e47518274a5e8) makes the test pass.

We need to investigate what causes this to fail; if the test-integration/test-integration-cli makes modifications to networking that causes this behavior

**Comment by thaJeztah** [comment]
This PR was created for debugging: https://github.com/moby/moby/pull/39562

**Comment by andrewhsu** [comment]
It seems that something in the `test-integration-cli` is mucking with the networking of the newly-compiled `dockerd` in a way that prevents the `docker-py` tests run immediately after it using the same `dockerd` to not have name resolution.

Discussing with @tonistiigi, a suggested safe way to isolate the networking changes would be to start a totally new instance of the newly-compiled `dockerd` to use to run the `docker-py` tests.

In this way, it would re-create the networking env to a known state to be sure the `docker-py` tests are starting fresh. Also, if the `docker-py` tests ever bonk out the new instance of the newly-compiled `dockerd` then it would only affect that daemon.

**Comment by thaJeztah** [comment]
Yes, that would work, and also was the idea to move to, once we could more easily modify the jenkins configuration (that, or even a fully separate job on Jenkins)
